### PR TITLE
notification sound + vibrate while in call

### LINKED ISSUE #1830: please add option to hear message tone while on a call
hi, please add a option to hear message tone while on a call.  I can not get the vibrate to work and I'm blind so I can't see the notification in the status bar this would be useful.  Your app is extremely accessible to the blind and with Google's screen reader talk back.  I would be willing to donate if you add this option thank you.

### PR DESCRIPTION
**Pull request by rhodey**
play notification sound and vibrate (if enabled) when new message received during a phone call.

Fixes #1830
// FREEBIE

### CODE REVIEW

File: src/org/thoughtcrime/securesms/notifications/MessageNotifier.java [hunk 1]
```diff
@@ -420,11 +424,28 @@ private static void setNotificationAlarms(Context context,
...
+      AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
+      if (audioManager.getRingerMode() == AudioManager.RINGER_MODE_NORMAL && ringtoneUri != null) {
+        Ringtone tone = RingtoneManager.getRingtone(context, Uri.parse(ringtone));
+        tone.setStreamType(AudioManager.STREAM_VOICE_CALL);
+        tone.play();
+      }
+
+      if (signal && vibrate) {
+        Vibrator vibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
+        vibrator.vibrate(new long[] {0, 250, 250, 250}, -1);
```
Comment (rhodey) [inline src/org/thoughtcrime/securesms/notifications/MessageNotifier.java]: default found in `NotificationManagerService` source, class is `@hide`n

Comment (mcginty) [inline src/org/thoughtcrime/securesms/notifications/MessageNotifier.java]: Do we not just want to use the vibrate pattern chosen in settings? Also, is vibrating while a phone is against your face really a feature people want?

**Comment by wp9015362** [comment]
> @mcginty wrote:
> 
> Also, is vibrating while a phone is against your face really a feature people want?

No, that would suck.

It could however be dealt with via an option if it weren't for this restrictive thing:

https://github.com/WhisperSystems/TextSecure/blob/master/contributing.md#development-ideology

...

**Comment by mcginty** [comment]
That "restrictive thing" is there *exactly* so we will never have a face vibrator option.

File: src/org/thoughtcrime/securesms/notifications/MessageNotifier.java (hunk 1 above)
Comment (rhodey) [inline src/org/thoughtcrime/securesms/notifications/MessageNotifier.java]: good point, I'll take out the vibrate. settings only allow led blink pattern ftr.

Comment (rhodey) [inline src/org/thoughtcrime/securesms/notifications/MessageNotifier.java]: good point, I'll take out the vibrate. settings only allow led blink pattern ftr.

**Pull request closed without merging**